opensearch_py_ml/ml_commons/ml_commons_client.py

```diff
@@ -198,3 +198,21 @@ def delete_model(self, model_id: str) -> object:
             method="DELETE",
             url=API_URL,
         )
+    
+    def search_model(self, algorithm_name: str = "") -> object:
```

Comment:
Thanks for raising the PR.

In search we can search multiple ways. I understand you were covering both ways from [here](https://opensearch.org/docs/2.6/ml-commons-plugin/api/#search-model). But these are just two difference example.

Let's think about how else customer can provide search query.

For example :

```
POST /_plugins/_ml/models/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "algorithm": "KMEANS"
          }
        }
      ]
    }
  }
}
```

I would like to cover all the possibilities we can search for model in this function.